### ISSUE TITLE
Add a way to close a Transport

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#361]
So that we can easily interrupt a dial or a listening.

**Comment by tomaka** [comment]
Should probably be named `close` so that users don't think that this also closes the `Transport` outputs, which is impossible to do in a generic way.

**Comment by tomaka** [comment]
No longer relevant after #565
